activerecord/test/cases/connection_adapters/schema_cache_test.rb

```diff
@@ -17,7 +17,11 @@ def test_primary_key
 
       def test_yaml_dump_and_load
         # Create an empty cache.
-        cache = SchemaCache.new @connection
+        cache = if defined?(PostgreSQLAdapter) && @connection.kind_of?(PostgreSQLAdapter)
```

Comment:
Adding a `schema_cache` method to the connection would make its usage here cleaner as well

Reply:
Make sense, just updated the PR accordingly.